pkg/core/resources.go

```diff
@@ -31,6 +31,11 @@ type (
 		Create(dag *ResourceGraph, param K) error
 	}
 
+	ConfigurableResource[K any] interface {
+		Resource
+		Configure(params K) error
+	}
+
```

Comment:
what do we do with this?

Reply:
I just find it helpful in the IDE — it helps me find all the other implementations of Configure, if I want to use those as a template.